src/dmd/errors.d

```diff
@@ -35,6 +35,37 @@ enum Classification : Color
     tip = Color.brightGreen,          /// for tip messages
 }
 
+
+enum AlwaysError
+{
+    no,
+    yes,
+}
+
+
+pragma(printf) private extern (C++) void noop(const ref Loc loc, const(char)* format, ...) {}
```

Comment:
note the `static if (__VERSION__ < 2092)` version guards for `pragma(printf)` below

Reply:
Ah. I reckoned they weren't needed for newer versions of the compiler but I forgot about bootstrapping.

Reply:
Correct